### PR TITLE
Make models know what libraries they require

### PR DESCRIPTION
**Pull request by peterNordin**
Also contains various cleanup in related code

### CODE REVIEW

File: HopsanGUI/GUIObjects/GUIContainerObject.cpp [hunk 1]
```diff
@@ -3499,3 +3491,22 @@ LogDataHandler2 *ContainerObject::getLogDataHandler()
     return mpModelWidget->getLogDataHandler();
     //return mpLogDataHandler;
 }
+
+QStringList ContainerObject::getRequiredComponentLibraries() const
+{
+    QStringList requiredLibraryPaths;
```
Comment (robbr48) [inline HopsanGUI/GUIObjects/GUIContainerObject.cpp]: Why is it called "Paths"? Shouldn't it be "requiredLibraryIds"?

File: Models/Animation Validation/MachinesValidation.hmf [hunk 2]
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <hopsanmodelfile hmfversion="0.4" hopsanguiversion="0.6.x_r7435" hopsancoreversion="0.6.x_r7431">
   <requirements>
-    <componentlibrary>DigitalHydraulicsLib</componentlibrary>
     <componentlibrary>HopsanDefaultComponentLibrary</componentlibrary>
```
Comment (robbr48) [inline Models/Animation Validation/MachinesValidation.hmf]: Why not use the library ID here, now that the default lib has one?

File: HopsanGUI/GUIObjects/GUIContainerObject.cpp (hunk 1 above)
Comment (peterNordin) [reply HopsanGUI/GUIObjects/GUIContainerObject.cpp]: Yes this is leftover from before introducing id. I forgot to change name. I will fix before merging

File: Models/Animation Validation/MachinesValidation.hmf (hunk 2 above)
Comment (peterNordin) [reply Models/Animation Validation/MachinesValidation.hmf]: I made this change before adding ID. Then I considered updating the models again. But decided not to (to much  work). This informatin is still valid for models saved with an older version of hopsan. The load code will show a debug message if the library name is not pressent. (Like before). Only difference is that now it wont warn for non needed libraries. 
The next time the models are updated (resaved) the new ID info will be added.

**Pull request merged**